core/src/main/java/org/apache/iceberg/jdbc/JdbcTableOperations.java

```diff
@@ -138,7 +138,7 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
       throw new UncheckedSQLException(e, "Database warning");
     } catch (SQLException e) {
       // SQLite doesn't set SQLState or throw SQLIntegrityConstraintViolationException
-      if (e.getMessage().contains("constraint failed")) {
+      if (e.getMessage() != null && e.getMessage().contains("constraint failed")) {
```

Comment:
please add tests that exercise each change here (similar to the test in https://github.com/apache/iceberg/pull/10069)